### ISSUE TITLE
Use Basic Event Hubs to reduce cost

### ISSUE THREAD

**Issue by gigamegawatts** [MSOpenTech/connectthedots#133]
I'm new to Azure, but it appears that AzurePrep creates 2 Standard Event Hubs (ehalerts and ehdevices), which after the trial period will cost a total of $44 per month.  

Can AzurePrep be changed to created Basic Event Hubs instead? 

That would still be $22 per month, so is it possible to redesign things so that only 1 Event Hub is needed?

**Comment by lt72** [comment]
assigned for execution

**Comment by dinarisio** [comment]
Done at ehconsoleUpd branch (will be merged soon) - now AzurePrep wil ask if you want to create ehalerts. When not, it uses ehdevices as output for creating SA jobs.

**Comment by dinarisio** [comment]
Merged with master.

**Comment by gigamegawatts** [comment]
Thanks for the quick response to my request.

I downloaded the updated code, and it works as described: 

AzurePrep displays a new prompt asking if I want to create ehalerts.   I selected No, and only the ehdevices Event Hub was created.  

The charts on the web site appear to be working correctly.  I'm not using the Stream Analytics feature, so I don't know if that is affected by this change.

As for the "Basic Event Hub" part of my request, and the effect on billing, I'm not sure if that is addressed by this change.  My terminology was probably not clear.

I was told a couple of days ago by Azure support that the Service Bus's Scale setting would have to be set the "Basic" - that changes the Event Hub from Standard to Basic.  I see that the Service Bus created by AzurePrep still has a Scale of Standard.  I tried manually changing the Scale to Basic, but the chart in the Web Site no longer appeared afterwards.  I changed it back to Standard, and the Web Site's charts are working again.

Is there any value, in terms of reduced cost, to changing AzurePrep to create a Basic service bus, rather than a Standard one?  If not, let me know and I'll close this issue as fixed.

**Comment by dinarisio** [comment]
Hello, it's true - I thought that the request is in reducing number of Event Hubs.
I've took a look and it appeared that one is not able to create any additional Consumer groups when Service Bus namespace has 'Basic' tier.
Meanwhile, our WebSite creates 'WebSite$date' and 'Local' consumer groups, and it is very reasonable. However, if you want to try scenario with 'Basic' tier and use the only '$Default' Consumer group - it is possible to change a code of Web site a bit (I've tested today). Also, you can push data to Event Hub without any code changes.

**Comment by gigamegawatts** [comment]
Hi,

OK, I looked at the code in Global.asx.cs, and I see what you mean.  I tried changing the code to only use the $Default consumer group, and that seems OK.

Unfortunately, I ran into a new pricing issue with the Web Site code. The new Azure SDK's Publish dialog requires an "App Service Plan", and after some trial-and-error I realized that the ConnectTheDots' Stream Analytics feature requires a more expensive Plan.  If I understand it correctly, it's a $50/month plan.

So, I guess I'd need to strip out the Streaming Analytics stuff out of the Web SIte code too, to make the whole thing affordable to a hobbyist.  

Thanks for reducing the # of Event Hubs, and the tips about the Consumer Groups.

**Comment by dinarisio** [comment]
There is not Stream Analytics code at the ConnectTheDotsWebSite, you can just avoid creation of these SA jobs during AzurePrep setup.

**Comment by gigamegawatts** [comment]
Hi,

Ok, I'm sure you're right - Stream Analytics was just a guess.

I know this is off-topic from the original issue with Event Hubs, so I don't want to waste your time on a different problem, but FYI here is the error message I was getting in Visual Studio's WebDeploy dialog:

"Error in /subscriptions/xxx/resourceGroup/Default-Web-EastUS/providers/microsoft.insights/components/mywebsitename: MissingSubscriptionRegistration: The subscription is not registered to use namespace microsoft.insights".

There are a long series of error messages like that - the part starting with MissingSubscriptionInformation is repeated in all error messages.

I didn't start seeing this problem until upgraded the Azure SDK on Monday night.  The Publish dialog has changed: it now contains new fields for "App Service Plan" and a "Resource Group".  When I selected the default App Service Plan (and one of the many Resource Group options), I got the above error.

I ended up fixing the problem by manually creating an "App Service Plan" in the Basic tier - the default one I was using when I got the above error is in the Free tier.

I couldn't figure out what "microsoft.insights" referred to: I guessed Stream Analytics, but maybe it was something else like the Alerts feature.  

Anyhow, that's just FYI, and for the benefit of other users who run into that error message: it's not very Google-able.

Thanks again,

Dan
